### ISSUE TITLE
Update example app for new develop-0.2 API

### ISSUE THREAD

**Issue by AndrewSB** [willowtreeapps/cordux#40]
I'm trying to update to the latest cordux API and I'm having trouble understanding what the `started` in 

```
public func coordinatorForTag(_ tag: String) -> (coordinator: AnyCoordinator, started: Bool)?
```

does. Does that mean my `SceneCoordinator` should be holding on to instances of it's child coordinators, and know whether or not they've been presented before, i.e. `started`?

Being able to see how the new API is used in the example app would be awesome 👍

**Comment by AndrewSB** [comment]
Or does it just signal whether or not the child coordinator has had `start()` called on it yet?

**Comment by ianterrell** [comment]
Yes — that is it. That is so that scene coordinators can "reuse" coordinators without having `start` called multiple times.

I definitely need to update the sample app... it is far behind my current thinking! Much of my thinking can be found in the issues though.

**Comment by AndrewSB** [comment]
would you recommend storing that `started` state inside the `SceneCoordinator`? Some sort of map from `RouteSegment` -> Coordinator instances?

checking out some of those issues

**Comment by ianterrell** [comment]
Yes — fundamentally I think it's the scene coordinator's responsibility to know whether or not to start each scene's coordinator.

**Comment by AndrewSB** [comment]
That makes sense, can you look over my implementation of my `AppCoordinator` and tell me where I'm fighting your framework?
Specifically, start and presentCoordinator?

```swift
       func start(route: Route?) {
            store.rootCoordinator = self

            guard let route = route, !route.isEmpty else {
                let initialRouteSegment: RouteSegment = User.Service.isLoggedIn ? .home : .auth
                return store.route(.push(initialRouteSegment.route())) // this will call changeScene
            }

        }

        func presentCoordinator(_ coordinator: AnyCoordinator?, completionHandler: @escaping () -> Void) {
            guard let coordinator = coordinator else { fatalError() }

            let filtered = childCoordinatorsPresented.values
                .map { coordinator, _ in coordinator }
                .filter { coordinator.route == $0.route }
            guard filtered.count == 1 else { fatalError() }

            let new = filtered.first!
            let newRouteSegment = RouteSegment(rawValue: new.route.components.last!)!

            switch childCoordinatorsPresented[newRouteSegment]! {
            case (_, true):  break
            case (_, false):
                new.start(route: newRouteSegment.route())
                childCoordinatorsPresented[newRouteSegment]!.1 = true
            }

            switchView(inContainerView: self.container, from: current.rootViewController, to: new.rootViewController, completion: completionHandler)
        }
```

(entire thing is [here](https://github.com/AndrewSB/wrytr/blob/master/wrytr/Molecules/Startup/AppCoordinator.swift))

**Comment by ianterrell** [comment]
Thanks — it is helpful to see how other people interpret the code without better documentation.

1. I don't think there is any need to store `childCoordinatorsPresented`. You only need to store the one currently presented, and that is done in `currentScene` [by `SceneCoordinator`](https://github.com/willowtreeapps/cordux/blob/develop-v0.2/Sources/SceneCoordinator.swift#L55).
1. There is no need to start the coordinator manually. `SceneCoordinator` is [handling that for you](https://github.com/willowtreeapps/cordux/blob/develop-v0.2/Sources/SceneCoordinator.swift#L56) already.

**Comment by AndrewSB** [comment]
How do I implement 
`coordinatorForTag(_ tag: String) -> (coordinator: AnyCoordinator, started: Bool)?` if I don't store `childCoordinatorsPresented`? I can return the coordinator, but I need something like `childCoordinatorsPresented` to hold onto the state on whether they've started or not, right?

**Comment by ianterrell** [comment]
If you're instantiating a new coordinator each time, then you could simply return `false` for `started`. Otherwise, yes, you would need to store something. That would be use case specific though. You can see an example of that in the [TabBarControllerCoordinator](https://github.com/willowtreeapps/cordux/blob/develop-v0.2/Sources/TabBarControllerCoordinator.swift), where they're stored and reused.
